packages/cli/src/constructs/check.ts

```diff
@@ -16,90 +16,215 @@ import { ConfigDefaultsGetter, makeConfigDefaultsGetter } from './check-config'
 import { validateDeprecatedDoubleCheck } from './internal/common-diagnostics'
 
+/**
+ * Base configuration properties for all check types.
+ * These properties are inherited by ApiCheck, BrowserCheck, and other check types.
+ * 
+ * @example
+ * ```typescript
+ * const commonCheckProps: Partial<CheckProps> = {
+ *   frequency: Frequency.EVERY_5M,
+ *   locations: ['us-east-1', 'eu-west-1'],
+ *   tags: ['production', 'api'],
+ *   alertChannels: [emailAlert, slackAlert],
+ *   retryStrategy: RetryStrategyBuilder.fixedStrategy({
+ *     maxRetries: 2,
+ *     baseBackoffSeconds: 60
+ *   })
+ * }
+ * ```
+ */
 export interface CheckProps {
   /**
-   *  The name of the check.
+   * The display name of the check. This will be shown in the Checkly dashboard.
+   * 
+   * @example "User API Health Check"
    */
   name: string
+  
   /**
-   *  Determines whether the check will run periodically or not after being deployed.
+   * Determines whether the check will run periodically after being deployed.
+   * Set to `false` to pause a check without deleting it.
+   * 
+   * @defaultValue true
    */
   activated?: boolean
+  
   /**
    * Determines if any notifications will be sent out when a check fails and/or recovers.
+   * Useful for temporarily silencing alerts during maintenance.
+   * 
+   * @defaultValue false
    */
   muted?: boolean
+  
   /**
    * Setting this to "true" will trigger a retry when a check fails from the failing region and another,
    * randomly selected region before marking the check as failed.
    * @deprecated Use {@link retryStrategy} instead.
    */
   doubleCheck?: boolean
+  
   /**
-   * Allows to invert the behaviour of when a check is considered to fail. Allows for validating error status like 404.
-   * This only applies to API Checks.
+   * Allows to invert the behaviour of when a check is considered to fail. 
+   * Useful for validating error status codes like 404 or 500.
+   * This only applies to API Checks. When set to true, the check passes when 
+   * it would normally fail, and fails when it would normally pass.
+   * 
+   * @defaultValue false
+   * @example
+   * ```typescript
+   * // Check that expects a 404 status - must set shouldFail: true
+   * shouldFail: true,
+   * request: {
+   *   method: 'GET',
+   *   url: 'https://api.example.com/nonexistent',
+   *   assertions: [AssertionBuilder.statusCode().equals(404)]
+   * }
+   * ```
    */
   shouldFail?: boolean
+  
   /**
-   * An array of one or more data center locations where to run this check. The supported regions are:
+   * An array of one or more data center locations where to run this check.
+   * The supported regions are:
    * us-east-1, us-east-2, us-west-1, us-west-2, ca-central-1, sa-east-1,
    * eu-west-1, eu-central-1, eu-west-2, eu-west-3, eu-north-1, eu-south-1, me-south-1,
    * ap-southeast-1, ap-northeast-1, ap-east-1, ap-southeast-2, ap-southeast-3, ap-northeast-2, ap-northeast-3,
    * ap-south-1, af-south-1
```

Comment:
Not sure if it's worth specifying all of these here tbh. I doubt we'll remember to keep it up to date, should it change.

Reply:
deleted.